### DISCUSSION TITLE
fatal: unable to access 'https://github.com/Akwailer/alx-system_engineering-devops.git/': The requested URL returned error: 403

### DISCUSSION THREAD

**Discussion by Akwailer** [community/community#57468] [New to GitHub]
### Select Topic Area

Question

### Body

fatal: unable to access 'https://github.com/Akwailer/alx-system_engineering-devops.git/': The requested URL returned error: 403

**Comment by airtower-luna** [comment]
403 Forbidden indicates your credentials weren't valid. You'll need to provide more details (full commands and error messages) to get more help than that.

**Comment by Akwailer** [comment]
In the terminal I cloned a repository, worked on a file and then I used git add to add the file to the commit log and when I did git commit, it worked fine. Finally, git push asks for username and password. I put those in correctly and every time I do this, it says the authentication error.

What is the cause of this problem and how can I fix it?

**Comment by airtower-luna** [comment]
`git add`, `git commit`, `git log`, etc. are purely *local* commands, they never even try to connect to a remote repository. `git push` does, because pushing to a remote is its purpose.

> Finally, git push asks for username and password. I put those in correctly and every time I do this, it says the authentication error.

You cannot use your password for `git push` or other Git operations with GitHub (but you still need it to log in). You need to [create a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) and enter that instead when Git asks for your password. When creating the token make sure to select the **repo** scope (for repository access, like pushing) and if you use Actions also **workflow** (required to modify workflow files).
